lib/src/api.ts

```diff
@@ -117,3 +140,97 @@ function parseSwapBackendInfo(response: SwapBackendResponse): {
     payinExtraId: response.payinExtraId,
   };
 }
+
+/**
+ * SELL *
+ **/
+
+export interface SellRequestPayload {
+  quoteId: string;
+  provider: string;
+  fromCurrency: string;
+  toCurrency: string;
+  refundAddress: string;
+  amountFrom: number;
+  amountTo: number;
+  nonce: string;
+}
+
+export interface SellResponsePayload {
+  sellId: string;
+  payinAddress: string;
+  createdAt: string;
+  providerFees: number;
+  referralFees: number;
+  payoutNetworkFees: number;
+  providerSig: {
+    payload: string;
+    signature: string;
+  };
+}
+
+const parseSellBackendInfo = (response: SellResponsePayload) => {
+  return {
+    sellId: response.sellId,
+    payinAddress: response.payinAddress,
+    providerSig: {
+      payload: response.providerSig.payload,
+      signature: response.providerSig.signature,
+    },
+  };
+};
+
+export async function retriveSellPayload(data: SellRequestPayload) {
+  // Make sure we are using the correct URL for the BE
+  setBackendUrlAsSell();
+
+  const request = {
+    quoteId: data.quoteId,
+    provider: data.provider,
+    fromCurrency: data.fromCurrency,
+    toCurrency: data.toCurrency,
+    refundAddress: data.refundAddress,
+    amountFrom: data.amountFrom,
+    amountTo: data.amountTo,
+    nonce: data.nonce,
+  };
+  const res = await axiosClient.post("", request);
+
+  return parseSellBackendInfo(res.data);
+}
+
+export async function decodeSellPayloadAndPost(
+  binaryPayload: string,
+  beData: BEData,
+  providerId: string,
+  customUrl?: string
+) {
+  const buyApiUrl =
+    customUrl ??
+    "https://buy.api.aws.prd.ldg-tech.com/sell/v1/forgeTransaction/offRamp";
```

Comment:
I am not sure why we do not use this customUrl as a unique "root" for all the call.

const buyApiUrl = `${customUrl ??
    "https://buy.api.aws.prd.ldg-tech.com/sell/v1"}/forgeTransaction/offRamp`;